app/javascript/components/widget-controller.jsx

```diff
@@ -33,234 +29,58 @@ const getWidgets = gql`
+        id: nextWidgetId,
+        cityName,
+      },
     });
   };
 
-  handleClicks = ({ target }) => {
-    const targetId = target.id;
-    if (targetId.startsWith('widget_')) {
-      const widgetId = parseInt(targetId.replace('widget_', ''), 10);
-      this.switchToPage(widgetId);
-    }
-  };
-
-  render() {
-    const { currentWidgetId, pausedTime } = this.state;
-
-    const fetchPolicy = currentWidgetId === 0 ? 'network-only' : 'cache-first';
-
-    const { cityName } = this.props;
-
-    return (
-      <Query
-        query={getWidgets}
-        variables={{ id: currentWidgetId, cityName }}
-        fetchPolicy={fetchPolicy}
-        onCompleted={response => {
-          const { location } = response;
-          const { widgets } = location;
-          const { enabled: enabledWidgets, byIdOrFirst: current } = widgets;
-
-          const index = enabledWidgets.findIndex(w => w.id === current.id);
-
-          const prevWidgetId =
-            enabledWidgets[mathMod(index - 1, enabledWidgets.length)].id;
-          const nextWidgetId =
-            enabledWidgets[mathMod(index + 1, enabledWidgets.length)].id;
-
-          this.setState({
-            currentWidgetId: current.id,
-            nextWidgetId,
-            prevWidgetId,
-            transitionTime: Date.now() + current.durationSeconds * 1000,
-          });
-        }}
-      >
-        {({ loading, error, data }) => {
-          if (loading) {
-            return (
-              <Wrapper>
-                <FullPanel>
-                  <LoadingMessage />
-                </FullPanel>
-                <SidePanel
-                  widgets={[]}
-                  selectedWidgetId={0}
-                  showWeather={false}
-                  totalTime={0}
-                  isPaused={false}
-                  cityName={cityName}
-                />
-              </Wrapper>
-            );
-          }
-
-          if (error) {
-            // eslint-disable-next-line
-            console.error(error);
-            return (
-              <Wrapper>
-                <FullPanel>
-                  <DisconnectedMessage />
-                </FullPanel>
-                <SidePanel
-                  widgets={[]}
-                  selectedWidgetId={0}
-                  showWeather={false}
-                  totalTime={0}
-                  isPaused={false}
-                  cityName={cityName}
-                />
-              </Wrapper>
-            );
-          }
-
-          const { location } = data;
-          const { widgets } = location;
-          const { enabled: enabledWidgets, byIdOrFirst: current } = widgets;
+  return (
+    <Query
+      query={getWidgets}
+      variables={{ id: newWidgetId, cityName }}
+      fetchPolicy={fetchPolicy}
+      onCompleted={response => {
+        const {
+          location: {
+            widgets: { byIdOrFirst: current },
+          },
+        } = response;
+        setNewWidgetId(current.id);
+      }}
+    >
```

Comment:
hmmmmm, this seems confusing. Why are we setting the newWidgetId onComplete? Is it possible that the widgetId that is returned the one that was selected? Will this cause the query to be fired twice on the first load?

Maybe what you want is the [useLazyQuery](https://www.apollographql.com/docs/react/api/react/hooks/#uselazyquery) hook?

Reply:
It does fire twice on the first load which I thought was strange.  Will look into useLazyQuery.

Reply:
> Why are we setting the newWidgetId onComplete?

Because on first load our widgetId is 0. Also a potential edge case (with the future admin page) is the queried widget could have been disabled, causing us to jump back to the first enabled widget.

Reply:
> It does fire twice on the first load which I thought was strange. Will look into useLazyQuery.

I would recommend reading and understanding the code before exploring alternatives. You can't assert the alternative is correct if you don't understand the problem.